packages/tiktok-video-element/tiktok-video-element.js

```diff
@@ -0,0 +1,324 @@
+const EMBED_BASE = 'https://www.tiktok.com/player/v1';
+
+const serialize = (props) => {
+  const boolToBinary = (props) => {
+    let p = {};
+    for (let key in props) {
+      let val = props[key];
+      if (val === true || val === '') p[key] = 1;
+      else if (val === false) p[key] = 0;
+      else if (val != null) p[key] = val;
+    }
+    return p;
+  };
+  return String(new URLSearchParams(boolToBinary(props)));
+};
+
+function serializeIframeUrl(attrs, id, config = {}) {
+  const params = {
+    controls: attrs.controls === undefined ? null : 1,
+    autoplay: attrs.autoplay,
+    loop: attrs.loop,
+    rel: 0,
+    ...config,
+  };
+
+  return `${EMBED_BASE}/${id}?${serialize(params)}`;
+}
+
+function getTemplateHTML(attrs, id, config = {}) {
+  const src = serializeIframeUrl(attrs, id, config);
+  const dataConfigAttr = config ? `data-config='${JSON.stringify(config).replace(/'/g, '&apos;')}'` : '';
+
+  return /*html*/ `
+    <style>
+      :host { display:inline-block;position:relative;width:100%;height:100%; }
+      iframe { position:absolute;top:0;left:0;width:100%;height:100%;border:0;}
+    </style>
+    <iframe src="${src}" ${dataConfigAttr} allow="autoplay; fullscreen" title="TikTok video"></iframe>
+  `;
+}
+
+class TikTokVideoElement extends (globalThis.HTMLElement ?? class {}) {
+  static getTemplateHTML = getTemplateHTML;
+  static shadowRootOptions = { mode: 'open' };
+
+  #muted = false;
+  #trueMuted = false; // Used to track actual mute state
+
+  static get observedAttributes() {
+    return ['src', 'controls', 'loop', 'autoplay', 'muted'];
+  }
+
+  static PlayerState = { INIT: -1, ENDED: 0, PLAYING: 1, PAUSED: 2, BUFFERING: 3 };
+
+  constructor() {
+    super();
+    this.attachShadow({ mode: 'open' });
+
+    this.loadComplete = this.#createPublicPromise();
+    this._currentTime = 0;
+    this.#muted = false;
+    this._paused = true;
+    this.playerState = TikTokVideoElement.PlayerState.INIT;
+
+    this._config = {};
+    this.#upgradeProperty?.('config');
+
+    this._onMessage = this.#handleMessage.bind(this);
+  }
+
+  connectedCallback() {
+    window.addEventListener('message', this._onMessage);
+    this.#render();
+  }
+
+  disconnectedCallback() {
+    window.removeEventListener('message', this._onMessage);
+  }
+
+  attributeChangedCallback(name, oldValue, newValue) {
+    if (name !== 'muted') {
+      this.#render();
+    } else {
+     if (newValue !== null && newValue !== 'false') {
+        this.mute();
+      } else {
+        this.unMute();
+      }
```

Comment:
this might sound weird but this should be removed because the `muted` attribute corresponds to the `video.defaultMuted` property. The `video.muted` property does not have any reflected attribute.

The `muted` attribute only has effect on the muted state when the video starts playing. 

https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/defaultMuted